native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.android.js

```diff
@@ -0,0 +1,241 @@
+    backgroundColor: colorGray50,
+    minHeight: spacingXxl + spacingBase,
+    flex: 1,
+  },
+  dismissableContainer: {
+    paddingVertical: spacingBase,
+    paddingHorizontal: spacingLg,
+    backgroundColor: colorGray50,
+  },
+  childrenContainer: {
+    paddingVertical: spacingSm,
+  },
+  container: {
+    display: 'flex',
+    flexDirection: 'row',
+  },
+  actionContainer: {
+    alignItems: 'flex-end',
+    justifyContent: 'center',
+    paddingHorizontal: spacingLg,
+    backgroundColor: colorGray50,
+  },
+  text: {
+    flex: 1,
+  },
+  icon: {
+    paddingRight: spacingSm,
+    paddingTop: spacingSm - spacingSm / spacingSm,
+  },
+  dismiss: {
+    color: colorBlue500,
+  },
+});
+
+const ALERT_TYPE_STYLES = {
+  [ALERT_TYPES.SUCCESS]: {
+    iconSource: 'tick-circle',
+    iconStyle: {
+      color: colorGreen500,
+    },
+  },
+  [ALERT_TYPES.WARN]: {
+    iconSource: 'information-circle',
+    iconStyle: {
+      color: colorYellow500,
+    },
+  },
+  [ALERT_TYPES.ERROR]: {
+    iconSource: 'information-circle',
+    iconStyle: {
+      color: colorRed500,
+    },
+  },
+  [ALERT_TYPES.NEUTRAL]: {
+    iconSource: 'information-circle',
+    iconStyle: {
+      color: colorGray500,
+    },
+  },
+};
+
+const BpkBannerAlert = props => {
+  const {
+    type,
+    message,
+    onDismiss,
+    onToggleExpanded,
+    dismissable,
+    expanded,
+    dismissButtonLabel,
+    toggleExpandedButtonLabel,
+    show,
+    style: userStyle,
+    animateOnEnter,
+    animateOnLeave,
+    children,
+    ...rest
+  } = props;
+
+  const expandable = children !== null;
+  const { iconSource, iconStyle } = ALERT_TYPE_STYLES[type] || {};
+
+  const dismissableContent = (
+    <TouchableNativeFeedback
+      accessibilityComponentType="button"
+      accessibilityLabel="dismiss"
+      background={TouchableNativeFeedback.SelectableBackgroundBorderless()}
+      useForeground
+      onPress={onDismiss}
+    >
+      <View style={styles.actionContainer}>
+        <BpkText textStyle="sm" emphasize style={styles.dismiss}>
+          DISMISS
+        </BpkText>
+      </View>
+    </TouchableNativeFeedback>
+  );
+  const expandableContent = (
+    <BpkAnimateHeight expanded={expanded}>
+      <View style={styles.childrenContainer}>{children}</View>
+    </BpkAnimateHeight>
+  );
+  const expandableIcon = (
+    <View style={styles.dismissableContainer}>
+      <BpkIcon
+        style={styles.icon}
+        icon={expanded ? 'chevron-up' : 'chevron-down'}
+        small
+      />
+    </View>
+  );
+  const mainContent = (
+    <View style={styles.container}>
+      <View style={styles.mainContainer}>
+        <View style={styles.container}>
+          <BpkIcon style={[styles.icon, iconStyle]} icon={iconSource} small />
+          <BpkText style={styles.text} textStyle="sm">
+            {message}
+          </BpkText>
+        </View>
+        {expandable && expandableContent}
+      </View>
+      {dismissable && dismissableContent}
```

Comment:
Isn't the order off here?

Reply:
Is not but the naming does not help, I've updated it